### ISSUE TITLE
How to get GIF play progress

### ISSUE THREAD

**Issue by songmeitao** [facebook/fresco#1694]
How to get GIF play progress???and ready progress???

**Comment by oprisnik** [comment]
Play progress: This is currently not possible. We're working on improving these APIs though.
Ready progress: A ControllerListener can be used according to http://frescolib.org/docs/animations.html

**Comment by songmeitao** [comment]
I still have a problem, my GIF in the listview play, how can fast play, or very smooth?When the slide stops quickly, the loading time of the GIF is very long.
How can I solve this problem?

**Comment by oprisnik** [comment]
We've released Fresco 1.4.0 with a completely new animation implementation.
In order to get the current animation duration you can use a controller listener to get the animation start time and subtract it from the current time. Something like

```
new BaseControllerListener(){
            @Override
            public void onFinalImageSet(String id, @Nullable Object imageInfo, @Nullable Animatable animatable) {
                if (animatable != null) {
                    AnimatedDrawable2 ad2 = (AnimatedDrawable2) animatable;
                    long startTimeMs = ad2.getStartTimeMs();
                    long animationTimeMs = SystemClock.uptimeMillis() - startTimeMs;
                }
            }
        }
```

**Comment by oprisnik** [comment]
You can also set an [animation listener](https://github.com/facebook/fresco/blob/master/animated-drawable/src/main/java/com/facebook/fresco/animation/drawable/AnimationListener.java) to get notified about these events.

You can use a `ControllerListener` to attach the `AnimationListener`:
```
Fresco.newDraweeControllerBuilder()
      .setImageRequest(imageRequest)
      .setAutoPlayAnimations(true)
      .setControllerListener(
              new BaseControllerListener() {
                  @Override
                  public void onFinalImageSet(
                          String id, 
                          @Nullable Object imageInfo, 
                          @Nullable Animatable animatable) {
                      if (animatable != null) {
                          AnimatedDrawable2 animatedDrawable = (AnimatedDrawable2) animatable;
                          animatedDrawable.setAnimationListener(yourAnimationListener);
                      }
                  }
              })
      .build()
```
